cart.js

```diff
@@ -0,0 +1,153 @@
+var productsList = getStorage(listKeys.productsListLocal) || {};
+var cartListElm = document.querySelector('.js-cart-list');
+var totalPriceElm = document.getElementById('total-price');
+var quantityCartElm = document.getElementById('quantity-cart');
+var cartEmptyElm = document.querySelector('.js-cart-empty');
+var cartNotEmptyElm = document.querySelector('.js-cart-not-empty');
+
+function renderCart() {
+  var cartList = getStorage(listKeys.cartList) || {};
+  var total = 0;
+  for (var cartId in cartList) {
+    var item = productsList[cartId];
+    var quantity = cartList[cartId].quantity;
+    var priceDiscount = (item.price - item.discount * item.price).toFixed(2);
+
+    cartListElm.innerHTML += `<li class="cart-item js-cart-item cart-item-${item.id}">
+      <div class="cart-item-left">
+        <div class="cart-img">
+          <img src="${item.thumbnail}" alt="" />
+        </div>
+        ${item.discount ? `<div class="badge badge-primary">${item.discount * 100 + '%'} </div>` : ""}
+        <div class="cart-info">
+          <div class="cart-info-top">
+            <h4 class="cart-name">T-Shirt Summer Vibe</h4>
+            <div class="cart-price">
+              ${item.discount ? `<p class="price-discount">${'$' + priceDiscount}</p>` : ""}
+              <p class="price-current">${'$' + item.price}</p>
+            </div>
+          </div>
+          <p class="cart-total-item">
+            Total: 
+            <span class="js-total-item" id="${"total-item-" +item.id}">
+              ${item.discount ? (priceDiscount * quantity).toFixed(2) : (item.price * quantity).toFixed(2)}
+            </span>
+          </p>
+        </div>
+      </div>
+      <div class="cart-option">
+        <div class="quantity">
+          <button 
+            data-id="${item.id}" 
+            class="js-btn-descrease">
+              -
+          </button>
+          <input type="number" id="${'quantity' + item.id}" disabled min="0" value="${quantity}"/>
+          <button 
+            data-id="${item.id}" 
+            class="js-btn-inscrease">
+              +
+          </button>
+        </div>
+        <button class="btn remove-btn js-remove-btn" data-id="${item.id}">Remove</button>
+      </div>
+    </li>`;
+     item.discount ? total += priceDiscount * quantity : total += item.price * quantity;
+  }
+
+  totalPriceElm.innerHTML = total.toFixed(2);
+}
+
+function updatePrice (cartItem, priceCurrent, id, value) {
+  var price = 0;
+  if(cartItem.discount) {
+    var priceDiscount = cartItem.price - cartItem.discount * cartItem.price;
+    price = (priceCurrent + priceDiscount * value).toFixed(2);
+  }
+  else {
+    price = (priceCurrent + cartItem.price * value).toFixed(2);
+  }
+  return price;
+}
+
+function updateQuantity(selector, value) {
```

Comment:
Đoạn này em có thể gán sự kiện vào button khi render ra ở hàm `renderCart` trên, do đó mình có thể tránh tạo thêm vòng for để gán click vào từng btn nữa

Reply:
anh Toàn Anh yêu cầu là dùng addEventListener, nên em dùng vòng for bên dưới để gắn event a